zerver/actions/user_groups.py

```diff
@@ -347,9 +347,21 @@ def do_send_delete_user_group_event(realm: Realm, user_group_id: int, realm_id:
     send_event(realm, event, active_user_ids(realm_id))
 
 
-def check_delete_user_group(
-    user_group_id: int, user_profile: UserProfile, *, acting_user: Optional[UserProfile]
-) -> None:
-    user_group = access_user_group_by_id(user_group_id, user_profile)
+def check_delete_user_group(user_group_id: int, *, acting_user: UserProfile) -> None:
+    user_group = access_user_group_by_id(user_group_id, acting_user)
+    RealmAuditLog.objects.create(
+        realm=acting_user.realm,
+        acting_user=acting_user,
+        event_type=RealmAuditLog.USER_GROUP_DELETED,
+        event_time=timezone_now(),
+        extra_data=orjson.dumps(
+            {
+                "name": user_group.name,
+                "realm_id": user_group.realm_id,
+                "description": user_group.description,
+                "is_system_group": user_group.is_system_group,
+            }
```

Comment:
It might be useful to list the set of members that were in it upon deletion, as well.

Reply:
Ideally we take the lock as part of checking the access control.